src/windows-emulator/minidump_loader.cpp

```diff
@@ -0,0 +1,758 @@
+        const auto& threads = dump_file->threads();
+        if (threads.empty())
+        {
+            win_emu.log.warn("No threads found in minidump\n");
+            return;
+        }
+
+        win_emu.log.info("Reconstructing threads: %zu threads\n", threads.size());
+
+        size_t success_count = 0;
+        size_t context_loaded_count = 0;
+
+        for (const auto& thread_info : threads)
+        {
+            try
+            {
+                emulator_thread thread(win_emu.memory);
+                thread.id = thread_info.thread_id;
+                thread.stack_base = thread_info.stack_start_of_memory_range;
+                thread.stack_size = thread_info.stack_data_size;
+
+                // Load CPU context if available
+                const bool context_loaded = load_thread_context(minidump_path, thread_info, thread.last_registers);
+                if (context_loaded)
+                {
+                    context_loaded_count++;
+                }
+
+                // Set TEB address if valid
+                if (thread_info.teb != 0)
+                {
+                    thread.teb = emulator_object<TEB64>(win_emu.memory);
+                    thread.teb->set_address(thread_info.teb);
+                }
+
+                win_emu.log.info("  Thread %u: TEB=0x%" PRIx64 ", stack=0x%" PRIx64 " (%u bytes), context=%s\n",
+                                 thread_info.thread_id, thread_info.teb, thread.stack_base, thread_info.stack_data_size,
+                                 context_loaded ? "loaded" : "unavailable");
+
+                win_emu.process.threads.store(std::move(thread));
+                success_count++;
+            }
+            catch (const std::exception& e)
+            {
+                win_emu.log.error("  Failed to reconstruct thread %u: %s\n", thread_info.thread_id, e.what());
+            }
+        }
+
+        // Set active thread to first available thread
+        if (success_count > 0)
+        {
+            auto& first_thread = win_emu.process.threads.begin()->second;
+            win_emu.process.active_thread = &first_thread;
+        }
+
+        win_emu.log.info("Thread reconstruction: %zu/%zu threads created, %zu with context\n", success_count,
+                         threads.size(), context_loaded_count);
+    }
+
+    void setup_peb_from_teb(windows_emulator& win_emu, const minidump::minidump_file* dump_file)
+    {
+        const auto& threads = dump_file->threads();
+        if (threads.empty())
+        {
+            win_emu.log.warn("No threads available for PEB setup\n");
+            return;
+        }
+
+        const auto& first_thread = threads[0];
+        if (first_thread.teb == 0)
+        {
+            win_emu.log.warn("Thread %u has null TEB address\n", first_thread.thread_id);
+            return;
+        }
+
+        try
+        {
+            // Read PEB address from TEB+0x60 (standard x64 TEB layout)
+            constexpr uint64_t teb_peb_offset = 0x60;
```

Comment:
https://github.com/momo5502/sogen/blob/main/src/common/platform/kernel_mapped.hpp#L459

```suggestion
            constexpr uint64_t teb_peb_offset = offsetof(TEB64, ProcessEnvironmentBlock);
```

comment is not needed anymore then.